pkg/asset/installconfig/platformcredscheck.go

```diff
@@ -35,7 +35,11 @@ func (a *PlatformCredsCheck) Generate(dependencies asset.Parents) error {
 	platform := ic.Config.Platform.Name()
 	switch platform {
 	case aws.Name:
-		_, err = awsconfig.GetSession()
+		ssn, err := awsconfig.GetSession()
+		if err != nil {
+			return err
+		}
+		err = awsconfig.ValidateCreds(ssn)
```

Comment:
This sets the `err` variable local to this case and not the `err` variable that is returned from the `Generate` function. In other words, errors from `awsconfig.ValidateCreds` will be ignored.

Personally, I would much rather see each of these cases check `err` and return when there is an error within the case block rather than fill out a shared `err` variable that is returned at the end of the function.